src/Plugin/QueueWorker/EloquaAppCloudDecisionQueueWorker.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Drupal\eloqua_app_cloud\Plugin\QueueWorker;
+
+use Drupal\Core\Queue\QueueWorkerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\eloqua_rest_api\Factory\ClientFactory;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueInterface;
+use Psr\Log\LoggerInterface;
+
+
+/**
+ * Give it 60 seconds?
+ *
+ * @property  logger
+ * @QueueWorker(
+ *  id = "eloqua_app_cloud_decision_queue_worker",
+ *  title = @Translation("The Eloqua AppCloud Queue worker for decisions."),
+ *  cron = {"time" = 60},
+ * )
+ */
+class EloquaAppCloudDecisionQueueWorker extends EloquaAppCloudQueueWorkerBase implements QueueWorkerInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal\eloqua_rest_api\Factory\ClientFactory definition.
+   *
+   * @var \Drupal\eloqua_rest_api\Factory\ClientFactory
+   */
+  protected $eloquaClientFactory;
+
+  /**
+   * @var QueueFactory
+   */
+  protected $queueFactory;
+
+  /** @var  LoggerInterface */
+  protected $logger;
+
+  /**
+   * Construct.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(
+    array $configuration, $plugin_id, $plugin_definition, ClientFactory $eloqua_client_factory, QueueFactory $queueFactory, LoggerInterface $logger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->eloquaClientFactory = $eloqua_client_factory;
+    $this->queueFactory = $queueFactory;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('eloqua.client_factory'),
+      $container->get('queue'),
+      $container->get('logger.channel.eloqua_app_cloud')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($queueItem) {
+    // The queue item may contain any number of contacts.
+    // We can only send 5000 at a time. If there are more then that we need to requeue the remainder.
+    // Either way, build a batch request and send it. Data items on the queue will contain an array
+    // of possible records. Iterate over the records and see what we need to do.
+    $records = &$queueItem->records;
+    // Splice off the first 5000 records.
+    // If there are any left requeue them for the next run through.
+    $chunk = array_splice($records, 0, 5000);
```

Comment:
Couple of comments here:
- That's some fancy PHP. I trust that it's doing what it ought to be doing.  I wonder if we need to make this a little more bullet-proof, though.  For example: if the Bulk API throws an exception below, we probably want to catch it, and re-queue any records in flight.  Maybe we can accomplish that without modifying this.  Just typing out loud...
- I wonder if it's necessary to make the 5000 number configurable.  To account for Eloqua slowness, or...

Reply:
Another possible failure mode: the thing that's running through the queue times out. What happens to the records that were in flight?

Reply:
I think the queue interface (or cron really) automatically requeues (is that a word?) if there is an exception in the worker. This pretty much covers any kind of failure.